s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,352 @@
+package pl.cyfronet.s4e.data.repository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.JdbcTemplate;
+import pl.cyfronet.s4e.bean.Product;
+import pl.cyfronet.s4e.bean.Scene;
+import pl.cyfronet.s4e.properties.DbProperties;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class SceneRepositoryImpl implements SceneRepositoryExt {
+    private final DbProperties dbProperties;
+    private Connection conn;
+    private int paramIndex;
+    private final JdbcTemplate jdbcTemplate;
+
+    public SceneRepositoryImpl(DbProperties dbProperties, JdbcTemplate jdbcTemplate) throws SQLException {
+        this.dbProperties = dbProperties;
+        this.conn = DriverManager.getConnection(dbProperties.getUrl(),
+                                                dbProperties.getUsername(),
+                                                dbProperties.getPassword());
+        this.paramIndex = 1;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public List<Scene> findAllByParamsMap(Map<String, Object> params) throws SQLException {
+        return jdbcTemplate.query(prepareQuery(params), (rs, rowNum) -> Scene.builder()
+                .id(rs.getLong("id"))
+                .product(Product.builder()
+                        .id(rs.getLong("product_id"))
+                        .build())
+                .build());
+    }
+
+    private String prepareQuery(Map<String, Object> params) throws SQLException {
+        PreparedStatement ps = conn.prepareStatement(prepareSqlQuery(params));
+        paramIndex = 1;
+        prepareParameters(ps, params);
+        log.trace(ps.toString());
+        return ps.toString();
+    }
+
+    private String prepareSqlQuery(Map<String, Object> params) {
+        StringBuilder resultQuery = new StringBuilder();
+        resultQuery.append("SELECT id, product_id FROM Scene WHERE true ");
```

Comment:
- `id`, `product_id` should be stored as const in SQL Model
- Do we have any SQL builder? It's easier to use (or to build if we don't have any).
- Why we are using this `walk around`? `WHERE true `

Reply:
1. what do you mean?
2. we search inside jsonb field. have to use native query, plain sql and no dsl. 
3. because there are no required parameters and i cant use any dsl, its easier for me to write  this condition then add "where" clause when there is any parameter.